src/differentials/thunks.jl

```diff
@@ -112,7 +112,7 @@ but it should do this more efficently than simply doing this directly.
 Most operations on an `InplaceableThunk` treat it just like a normal `Thunk`;
 and destroy its inplacability.
 """
-struct InplaceableThunk{T<:Thunk, F} <: AbstractThunk
+struct InplaceableThunkedTangent{T<:Thunk, F} <: AbstractThunk
```

Comment:
This seems too long and I've changed it only to highlight the inconsistency we get if we keep it as `InplaceableThunk`.

Reply:
Maybe we shouldn't use the Tangent suffix for Thunks?

Reply:
How about `InplaceableTangent`?

Reply:
Two reasons against that:
- hides arguably the more important info for handling this object (that it is a thunk)
- many tangents are "inplaceable" without being this type (so having something named InplaceableTangent is confusing)

Personally I prefer both the current name and the more verbose name

Reply:
Ok, I've changed it back to the current name. Are there any other open questions or are we happy with this? I'll go ahead and open follow up PRs everywhere